airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/DisableActivityImpl.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.config.JobConfig.ConfigType;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardSync.Status;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.scheduler.models.Job;
+import io.airbyte.scheduler.models.JobStatus;
+import io.airbyte.scheduler.persistence.JobPersistence;
+import io.airbyte.workers.temporal.exception.RetryableException;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@AllArgsConstructor
+@Slf4j
+public class DisableActivityImpl implements DisableActivity {
+
+  @VisibleForTesting
+  public static final int MAX_FAILURE_JOBS_IN_A_ROW = 100;
+  @VisibleForTesting
+  public static final int MAX_DAYS_OF_STRAIGHT_FAILURE = 14;
+
+  private final ConfigRepository configRepository;
+  private JobPersistence jobPersistence;
+
+  // if no successful sync jobs in the last MAX_FAILURE_JOBS_IN_A_ROW job attempts or the last
+  // MAX_DAYS_OF_STRAIGHT_FAILURE days (minimum 1 job attempt): disable connection to prevent wasting
+  // resources
+
+  @Override
+  public void disableConnection(final DisableActivityInput input) {
+    try {
+      // lists job in descending order by created_at
+      final List<Job> jobs = jobPersistence.listJobs(ConfigType.SYNC,
```

Comment:
I think that we should include the reset connection type here. To be check with product.